sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -677,3 +677,10 @@ def test_categories(density, drop):
 @pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
 def test_encoders_has_categorical_tags(Encoder):
     assert 'categorical' in Encoder()._get_tags()['X_types']
+
+
+def test_ordinal_encoder_deprecated_categories_auto_with_string_labels():
+    enc = OrdinalEncoder()
+    X = [['Male', 1], ['Female', 3], ['Female', 2]]
```

Comment:
If there is not another test doing so: Please include a 2-digit number to ensure that integers do not land up being encoded as strings.

Please check multiple ways strings may be presented, including a dataframe

Reply:
There were some test for double digit already. I am adding ways of presenting strings.